tests/support/test.tcl

```diff
@@ -126,11 +126,11 @@ proc wait_for_condition {maxtries delay e _else_ elsescript} {
     }
 }
 
-# try to match a value to a list of patterns that is either regex, or plain sub-string
-proc search_pattern_list {value pattern_list {substr false}} {
+# try to match a value to a list of patterns that are either regex (starts with "/") or plain string.
+proc search_pattern_list {value pattern_list} {
     set n 0
     foreach el $pattern_list {
-        if {[string length $el] > 0 && ((!$substr && [regexp -- $el $value]) || ($substr && [string match $el $value]))} {
+        if {[string length $el] > 0 && (([string match /* $el] && [regexp -- [string range $el 1 end] $value]) || [string equal $el $value])} {
```

Comment:
using `string match /*` to check if the first char is `/` seems like taking down a fly with a nuke.

Reply:
fixed in https://github.com/redis/redis/pull/10741/commits/032cc08f69cf70a5a2394608971ef792b70e93e4